UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -761,39 +767,70 @@ private void MenuItem_RunBuiltinScript_SubmenuOpened(object sender, RoutedEventA
+
+            await t;
+        }
+
+        private async Task RunScriptNow(string path)
         {
             Debug.WriteLine(path);
 
-            CommandBox.Text = "Running " + System.IO.Path.GetFileName(path) + " ...";
+            Dispatcher.Invoke(() => CommandBox.Text = "Running " + System.IO.Path.GetFileName(path) + " ...");
             try
             {
                 using (var loader = new InteractiveAssemblyLoader())
                 {
                     loader.RegisterDependency(typeof(UndertaleObject).GetTypeInfo().Assembly);
+                    loader.RegisterDependency(GetType().GetTypeInfo().Assembly);
+                    loader.RegisterDependency(typeof(JsonConvert).GetTypeInfo().Assembly);
 
                     var script = CSharpScript.Create<object>(File.ReadAllText(path), ScriptOptions.Default
                         .AddImports("UndertaleModLib", "UndertaleModLib.Models", "UndertaleModLib.Decompiler", "UndertaleModLib.Scripting")
-                        .AddImports("System", "System.IO", "System.Collections.Generic", "System.Text.RegularExpressions")
+                        .AddImports("UndertaleModTool", "System", "System.IO", "System.Collections.Generic", "System.Text.RegularExpressions")
                         .AddReferences(Program.GetAssemblyMetadata(typeof(UndertaleObject).GetTypeInfo().Assembly))
+                        .AddReferences(GetType().GetTypeInfo().Assembly)
+                        .AddReferences(Program.GetAssemblyMetadata(typeof(JsonConvert).GetTypeInfo().Assembly))
                         .AddReferences(typeof(System.Text.RegularExpressions.Regex).GetTypeInfo().Assembly),
                         typeof(IScriptInterface), loader);
 
                     object result = (await script.RunAsync(this)).ReturnValue;
-                    CommandBox.Text = result != null ? result.ToString() : System.IO.Path.GetFileName(path) + " finished!";
+                    Dispatcher.Invoke(() => CommandBox.Text = result != null ? result.ToString() : System.IO.Path.GetFileName(path) + " finished!");
                 }
             }
             catch (CompilationErrorException exc)
             {
-                Debug.WriteLine(exc.ToString());
-                CommandBox.Text = exc.Message;
+                Console.WriteLine(exc.ToString());
+                Dispatcher.Invoke(() => CommandBox.Text = exc.Message);
                 MessageBox.Show(exc.Message, "Script compile error", MessageBoxButton.OK, MessageBoxImage.Error);
             }
             catch (Exception exc)
             {
-                Debug.WriteLine(exc.ToString());
-                CommandBox.Text = exc.Message;
-                MessageBox.Show(exc.Message, "Script error", MessageBoxButton.OK, MessageBoxImage.Error);
+                Console.WriteLine(exc.ToString());
+                Dispatcher.Invoke(() => CommandBox.Text = exc.Message);
+                ScriptMessage("Script Error: " + exc.Message + "\n" + exc.ToString());
```

Comment:
I preferred the MessageBox tbh, it grabs attention more than the output window

Reply:
Huh? It still makes a MessageBox.

Reply:
Ah, right, my bad, ScriptMessage is still a MessageBox 😂 It doesn't have the error icon though.

Reply:
True, could probably fix that.

Reply:
Follow-up, I will fix it so it won't use ScriptMessage, but use MessageBox.Show again. That was a temporary thing I was doing for testing.